pkg/util/push/otel.go

```diff
@@ -109,16 +111,44 @@ func OTLPHandler(
 			return body, err
 		}
 
+		log, ctx := spanlogger.New(ctx, "Distributor.OTLPHandler.decodeAndConvert")
+		defer log.Span.Finish()
+
+		log.SetTag("content_type", contentType)
+		log.SetTag("content_encoding", contentEncoding)
+		log.SetTag("content_length", r.ContentLength)
+
 		otlpReq, err := decoderFunc(body)
 		if err != nil {
 			return body, err
 		}
 
+		level.Debug(log).Log("msg", "decoding complete")
```

Comment:
nit: improve message to indicate that we're not done yet despite "complete" in the message.

```suggestion
		level.Debug(log).Log("msg", "decoding complete, starting conversion")
```